synapse/lib/node.py

```diff
@@ -47,6 +51,25 @@ def __init__(self, snap, sode, bylayer=None):
         if self.nodedata is None:
             self.nodedata = {}
 
+    def getStorNodes(self):
+        '''
+        Return a list of the raw storage nodes for each layer.
+        '''
+        s = copy.deepcopy(self.sodes)
+        # snap has layers reversed as a node construction optimization
+        s.reverse()
+        return s
+
+    def getByLayer(self):
+        '''
+        Return a dictionary that translates the node's bylayer dict to a primitive.
+        '''
+        ndef = self.bylayer.get('ndef').iden
+        tags = {t: l.iden for (t, l) in self.bylayer.get('tags', {}).items()}
+        props = {p: l.iden for (p, l) in self.bylayer.get('props', {}).items()}
+        tagprops = {p: l.iden for (p, l) in self.bylayer.get('tagprops', {}).items()}
+        return {'ndef': ndef, 'props': props, 'tags': tags, 'tagprops': tagprops}
```

Comment:
'ndef' is an iden not an ndef

Reply:
The dict is a way to lookup which layer (iden) each of the things came from, so for this purpose the `ndef` key will point to a layer `iden`